homeassistant/components/sensor/statistics.py

```diff
@@ -111,8 +111,7 @@ def _add_state_to_queue(self, new_state):
         try:
             self.states.append(float(new_state.state))
             if self._max_age is not None:
-                now = dt_util.utcnow()
-                self.ages.append(now)
+                self.ages.append(new_state.last_updated)
```

Comment:
Do you even need a separate array for ages? You can just read the states from `self.states` ?

Reply:
Yes, that's correct. I opted for the lightest-touch update.

As an alternative, one could simply append `new_state` each time, and then specifically refer to either `self.state.states` or `self.state.last_updated` elsewhere in the code.

Reply:
As a follow up to this: implementation is not as simple as that. The states are stored in 
[`deque` objects](https://docs.python.org/2/library/collections.html#collections.deque), presumably for speed of appending and popping. These are list-like objects and as such don't support accessing by attributes such as `self.states.state` and `self.states.last_updated`.

Given this, for now I think the current implementation is fine: store ages / states in their own `deque` objects.